docs/modules/ROOT/pages/authenticationclass.adoc

```diff
@@ -1,3 +1,48 @@
 = `AuthenticationClass`
 
-TODO
+`AuthenticationClass` is a CRD describing a generic authentication method like LDAP or Kerberos.
+Multiple operators use this CRD as a way to express the authentication of the product.
+
+At the moment the following Authentication providers are supported:
+
+* <<LDAP>>
+
+== LDAP
+A very simple `AuthenticationClass` with LDAP Authentication looks like this:
+
+[source,yaml]
+----
+include::example$authenticationclass-ldap-simple.yaml[]
+----
+<1> The hostname of the LDAP server without any protocol or port
+<2> The port of the LDAP server. If TLS is used it defaults to `636` otherwise to `389`
+<3> An optional searchBase where the users should be searched
+
+=== OpenLDAP
+Here is an example that is tuned for an OpenLDAP LDAP server:
+
+[source,yaml]
+----
+include::example$authenticationclass-ldap-openldap-simple.yaml[]
+----
+<1> The name of the xref:secret-operator::secretclass.adoc[] providing the bind credentials (username and password). Must match the name of the xref:secret-operator::secretclass.adoc[] in this example in ②
```

Comment:
Comment refers to all doc links, I think it would make the docs more readable to provide text for the links here

`The name of the xref:secret-operator::secretclass.adoc[SecretClass]`

Reply:
If i render the antora docs everything works as intended
![image](https://user-images.githubusercontent.com/29303194/160853011-484e9a33-0de8-4893-add9-938894daa38a.png)

Reply:
Ah, ok. I didn't render Antora, I just checked the preview in my Asciidoc renderer. That is fine then.